daemon/main.cpp

```diff
@@ -196,8 +222,7 @@ main(int argc, char* argv[])
 
     if (genconfigOnly)
     {
-      llarp::ensureConfig(
-          llarp::GetDefaultDataDir(), llarp::GetDefaultConfigPath(), overwrite, opts.isRelay);
+      llarp::ensureConfig(basedir, fname, overwrite, opts.isRelay);
```

Comment:
I can't help but feel like this is wholly unrelated to exit changes, and yet I'm okay with it being in here...